src/CamelKJSONUtils.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+import * as http from "http";
+
+export const proxyURLSetting = 'camelk.integrations.proxyURL';
+export const proxyNamespaceSetting = 'camelk.integrations.proxyNamespace';
+
+const BACKSLASH = '\u005C';
+
+export function createCamelKRestURL() : string {
+	let server = vscode.workspace.getConfiguration().get(proxyURLSetting);
+	let namespace = vscode.workspace.getConfiguration().get(proxyNamespaceSetting);
+	let outputUrl = server + "/apis/camel.apache.org/v1alpha1/namespaces/" + namespace + "/integrations";
+	return outputUrl;
+}
+
+export function createCamelKDeleteRestURL(integrationName:string) : string {
+	let baseUrl = createCamelKRestURL();
+	let outputUrl = baseUrl + "/" + integrationName;
+	return outputUrl;
+}
+
+export function createCamelKPodLogURL(podName:string) : string {
+	let server = vscode.workspace.getConfiguration().get(proxyURLSetting);
+	let namespace = vscode.workspace.getConfiguration().get(proxyNamespaceSetting);
+	let outputUrl = server + "/api/v1/namespaces/" + namespace + "/pods/" + podName + "/log";
+	return outputUrl;
+}
+
+export function createCamelKGetPodsURL() : string {
+	let server = vscode.workspace.getConfiguration().get(proxyURLSetting);
+	let namespace = vscode.workspace.getConfiguration().get(proxyNamespaceSetting);
+	let outputUrl = server + "/api/v1/namespaces/" + namespace + "/pods";
+	return outputUrl;
+}
+
+function cleanText(text:string) : Promise<string> {
+	return new Promise( (resolve, reject) => {
+		if (text) {
+			let reg = new RegExp('(?:\\\")*');
```

Comment:
why is it needed to c"clean" text with specific code? It is a non-standard encoded text? where is the encoding done on the other side so that we can check that we covered all cases?

Reply:
When wrapping the file contents in the JSON for deployment to Kubernetes, I discovered some interesting issues when using JSON.stringify that had to be corrected, thus the RegEx dancing. See lines 75 and 102 where the cleanText function is called. This is just a sanitizing pass to ensure that we aren't doubling up the slashes, which was happening.

Reply:
I still don't get why JSON.stringify is returning invalid strings. it is a bug on what the server is returning? or on the way we are decoding the response?

Reply:
For some reason I saw all sorts of doubling of slashes between grabbing the file content and using JSON. Stringify -- WHY that happened, I don't know. But this little cleanup pass does the trick and I am able to package Groovy, Java, JavaScript, and XML files in the YAML format they need for deploying an integration.

Reply:
@lburgazzoli Do you have an idea why the provided json contains several double-slash, which seems to be invalid as it needs to be processed on client side. is it a specific encoding which needs to be decoded? or is ti a manual task which is doubling the escape so that we can dedouble them also in our codebase?

Reply:
sorry, which json ?

Reply:
hum i thought that it would have been somethign specific to the Camel-K call.
by re-reading code, it seems that ti is not. In fact, it seems that it is because Json.stringify is called instead of JSON.parse or simply reusing the reference (if I understand well this Stackoverflow answer https://stackoverflow.com/questions/16507858/json-stringify-escapes-double-quotes-every-time-when-stringified )

Reply:
Discovered we no longer needed the cleanText method at all. After removing the JSON.stringify from the file text, all the extra slashes went away by themselves. The cleanText just ended up being unnecessary in the end.